smartsim/entity/files.py

```diff
@@ -130,3 +130,122 @@ def _check_path(self, file_path):
+
+    @classmethod
+    def from_list_paths(cls, path_list, dir_contents_to_base=False):
+        """Given a list of absolute paths to files and dirs, create and return
+        a TaggedFilesHierarchy instance representing the file hierarchy of
+        tagged files. All files in the path list will be placed in the base of
+        the file hierarchy.
+
+        :param path_list: list of absolute paths to tagged files or dirs
+                          containing tagged files
+        :type path_list: list[str]
+        :param dir_contents_to_base: When a top level dir is encountered, if
+                                     this value is truthy, files in the dir are
+                                     put into the base hierarchy level.
+                                     Otherwise, a new sub level is created for
+                                     the dir
+        :type dir_contents_to_base: bool
+        :return: A built tagged file hierarchy for the given files
+        :rtype: TaggedFilesHierarchy
+        """
+        tagged_file_hierarchy = cls()
+        if dir_contents_to_base:
+            new_paths = []
+            for path in path_list:
+                if os.path.isdir(path):
+                    new_paths += [os.path.join(path, file) for file in os.listdir(path)]
+                else:
+                    new_paths.append(path)
+            path_list = new_paths
+        tagged_file_hierarchy._add_paths(path_list)
+        return tagged_file_hierarchy
+
+    def _add_file(self, file):
+        """Add a file to the current level in the file hierarchy
+
+        :param file: absoute path to a file to add to the hierarchy
+        :type file: str
+        """
+        self.files.add(file)
+
+    def _add_dir(self, dir):
+        """Add a dir contianing tagged files by creating a new sub level in the
+        tagged file hierarchy. All paths within the directroy are added to the
+        the new level sub level tagged file hierarchy
+
+        :param dir: absoute path to a dir to add to the hierarchy
+        :type dir: str
+        """
+        tagged_file_hierarchy = TaggedFilesHierarchy(
+            path.join(self.base, path.basename(dir))
+        )
+        tagged_file_hierarchy._add_paths(
+            [path.join(dir, file) for file in os.listdir(dir)]
+        )
+        self.dirs.add(tagged_file_hierarchy)
+
+    def _add_paths(self, paths):
+        """Takes a list of paths and iterates over it, determining if each
+        path is to a file or a dir and then appropriatly adding it to the
+        TaggedFilesHierarchy.
+
+        :param paths: list of paths to files or dirs to add to the hierarchy
+        :type paths: list[str]
+        :raises SSConfigError: if link to dir is found or path does not exist
+        """
+        for path in paths:
+            path = os.path.abspath(path)
+            if os.path.isdir(path):
+                if os.path.islink(path):
+                    raise SSConfigError(
```

Comment:
``ValueError``

Reply:
Resolved in https://github.com/CrayLabs/SmartSim/pull/119/commits/d92cdfd634d862d2e266fd6d364fd784e37df313